src/apps/material/helper.ts

```diff
@@ -146,13 +147,101 @@ export const totalBranchesHaveMaterial = (
   }).length;
 };
 
-export const getInfomediaId = (manifestation: Manifestation) => {
-  const access = manifestation?.access || [];
-  return access.reduce<string | null>((acc, curr) => {
-    const { __typename: type } = curr;
-    if (type === "InfomediaService") {
-      return curr.id;
+export const getInfomediaIds = (manifestations: Manifestation[]) => {
+  const infomediaIds = manifestations
+    .map((manifestation) =>
+      manifestation.access.map((currentAccess) => {
+        return currentAccess.__typename === "InfomediaService"
+          ? currentAccess.id
+          : null;
+      })
+    )
+    .flat();
+  return compact(infomediaIds);
+};
+
+export const getAllUniqueMaterialTypes = (manifestations: Manifestation[]) => {
+  const allMaterialTypes = manifestations
+    .map((manifest) => manifest.materialTypes.map((type) => type.specific))
+    .flat();
+  return uniq(allMaterialTypes);
+};
+
+export const divideManifestationsByMaterialType = (
+  manifestations: Manifestation[]
+) => {
+  const uniqueMaterialTypes = getAllUniqueMaterialTypes(manifestations);
+  const dividedManifestationsArrays = uniqueMaterialTypes.map(
+    (uniqueMaterialType) => {
+      return manifestations.filter((manifest) => {
+        const manifestationMaterialTypes = manifest.materialTypes.map(
+          (materialType) => materialType.specific
+        );
+        return manifestationMaterialTypes.includes(uniqueMaterialType);
+      });
     }
-    return acc;
-  }, null);
+  );
+  const dividedManifestationsObject: { [key: string]: Manifestation[] } =
+    dividedManifestationsArrays.reduce((result, current, index) => {
```

Comment:
Please note that you can also use the generic coming from ts reduce:
`dividedManifestationsArrays.reduce<{ [key: string]: Manifestation[] }>( `
or
`dividedManifestationsArrays.reduce<Record<string, Manifestation[]>>( `
And then you do not need to define the `const dividedManifestationsObject` but you can return it right away.